op-e2e/sequencer_failover_setup.go

```diff
@@ -345,22 +359,24 @@ func sequencerActive(t *testing.T, ctx context.Context, rollupClient *sources.Ro
 	return active
 }
 
-func findAvailablePort(t *testing.T) int {
+func findAvailablePort(t *testing.T, floor int) int {
 	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+	i := floor + 1
 	defer cancel()
 	for {
 		select {
 		case <-ctx.Done():
 			t.Error("Failed to find available port")
 		default:
 			// private / ephemeral ports are in the range 49152-65535
-			port := rand.Intn(65535-49152) + 49152
+			port := i
 			addr := fmt.Sprintf("127.0.0.1:%d", port)
 			l, err := net.Listen("tcp", addr)
 			if err == nil {
 				l.Close() // Close the listener and return the port if it's available
 				return port
 			}
+			i++
```

Comment:
This is doomed to fail unfortunately.  The port is available during this check but may not be by the time the actual service tries to open it.  The only safe way to select an available port is to specify 0 when selecting the port in the service.

I have seen people try every variation of this kind of method and it always winds up being intermittent. You'd think it would be very rare, but it always winds up being way more common than anyone sane person would expect.

Reply:
```
// it's unfortunate that it is not possible to pass 0 as consensus port and get back the actual assigned port from raft implementation.
// So we find an available port and pass it in to avoid test flakiness (avoid port already in use error).
```

Unfortunately current raft implementation does not allow us to do that, we could either, I propose that we
1. upstream a PR to add the feature to raft implementation (might take some time)
2. and in the mean time, accept the fact it will happen sometimes (shouldn't be that often, we can increase the random port range to reduce collision as an intermediate measure)

Reply:
We could always fork raft while it is getting accepted into upstream

Reply:
Actually if you can't use 0 the only other option is to pick a random port, start the service and detect if it fails because of a port conflict and retry until you get a free port.  The key thing is that you're starting the actual service so if the port is free it keeps it open rather than closing and hoping it's still free when the service starts again.

Using docker also works as it can assign random local ports and let you know what they are but is pretty heavy handed.